bftengine/src/preprocessor/RequestProcessingInfo.cpp

```diff
@@ -85,4 +85,9 @@ unique_ptr<MessageBase> RequestProcessingInfo::convertClientPreProcessToClientMs
   return clientPreProcessReqMsg_->convertToClientRequestMsg(resetPreProcessFlag);
 }
 
+void RequestProcessingInfo::freeClientMsg() {
+  delete clientPreProcessReqMsg_;
+  clientPreProcessReqMsg_ = nullptr;
```

Comment:
Similar to my comment above, can we make `clientPreProcessReqMsg_` a smart pointer instead of doing the free ourselves?

Reply:
As above.